.gitlab-ci.yml

```
@@ -64,6 +65,28 @@ test:unit:
     paths:
       - coverage
 
+test:docs-links:
+  stage: test
+  except:
+    - /^saas-[a-zA-Z0-9.]+$/
+  before_script:
+    - apk add --no-cache curl
+  script:
+    - links=$(grep -r docs.mender.io src/ | grep -v snapshots | sed -e 's,\${docsVersion},'${DOCS_VERSION}'/,g' | egrep -o 'https?://[^ `]+' | sort | uniq)
+    - error=0
+    - for url in $links; do
+    -   if ! curl --output /dev/null --silent --head --fail $url; then
+    -     error=1
+    -     echo "[ko] $url"
+    -   else
+    -     echo "[ok] $url"
+    -   fi
+    - done
+    - if [ $error -gt 0 ]; then
+    -   exit 1
+    - fi
+  allow_failure: true
```

Why you want to allow failure?

---

I mean, I know that right now they are some broken links because we have not yet updated mender-docs, but leaving the `allow_failure: true` in the Pipeline I think defeats its purpose.

---

that's because right now it would fail, and I couldn't come up with a "smart way" to avoid it.
From my point of view, this is more a "safety net" and I'll take care of having that job green before deploying this version to production. Making it conditional requires additional effort, not sure it makes sense. Any suggestion?

---

I was not thinking about making it conditional, but just let it fail and force-merge this PR (and others that might happen in exceptional situations). But now that you say so, I had an idea:

We could have this job to use `DOCS_VERSION=development` (and except on branch staging), and a job cloned from this one to run for staging branch with `DOCS_VERSION=hosted`. That will let the gui repo evolve following master docs, but in staging always double check that links are good before production.

If this is a one time thing, and we are removing the `allow_failure` later, then I don't have objections in leaving it as is.

---

Yup, done!